src/django_pg_migration_tools/management/commands/migrate_with_timeouts.py

```diff
@@ -31,6 +54,111 @@ def add_arguments(self, parser: base.CommandParser) -> None:
             required=False,
             help="Value to set as statement_timeout in milliseconds.",
         )
+        parser.add_argument(
+            "--retry-callback--path",
+            dest="retry_callback_path",
+            type=str,
+            required=False,
+            help=(
+                "If retries are set, this argument can be used with the dotted path "
+                "to a function to be called between retries. The function must "
+                "follow this signature: f(retry_state) where `retry_state` "
+                "is the dataclass: "
+                "django_pg_migration_tools.management.commands.migrate_with_timeouts.RetryState."
+                "This callback can be useful for calling loggers after each "
+                "retry."
+            ),
+        )
+        parser.add_argument(
+            "--lock-timeout-max-retries",
+            dest="lock_timeout_max_retries",
+            type=int,
+            required=False,
+            default=0,
+            help=(
+                "How many times to retry after a lock timeout happened. "
+                "Defaults to zero, which means no retries - the migration "
+                "fails immediatelly upon lock timeout."
+            ),
+        )
+        parser.add_argument(
+            "--lock-timeout-retry-exp",
+            dest="lock_timeout_retry_exp",
+            type=int,
+            required=False,
+            default=2,
+            help=(
+                "The value for the exponent for retry backoff propagation. "
```

Comment:
🐼  "propagation" doesn't seem like the right word, but it's not clear to me what the right word is. Would "delay" work?

Reply:
03d3ed2613043d4dc358fc8d7e368d515dafb016